app/models/manageiq/providers/redfish/physical_infra_manager/physical_server/provisioning.rb

```diff
@@ -0,0 +1,41 @@
+module ManageIQ::Providers::Redfish
+  module PhysicalInfraManager::PhysicalServer::Provisioning
+    def deploy_pxe_config(pxe_image, customization_template)
+      with_provider_object do |system|
+        if (macs = mac_addresses(system)).empty?
+          raise MiqException::MiqProvisionError, 'at least one MAC address is needed for provisioning'
+        end
+        macs.each { |mac| pxe_image.pxe_server.create_provisioning_files(pxe_image, mac, nil, customization_template) }
+      end
+    end
+
+    def reboot_using_pxe
+      with_provider_object do |system|
+        response = system.patch(
+          :payload => {
+            'Boot' => {
+              'BootSourceOverrideEnabled' => 'Once',
+              'BootSourceOverrideTarget'  => 'Pxe'
+            }
+          }
+        )
+        raise MiqException::MiqProvisionError, 'Cannot override boot order' if response.status >= 400
+      end
+      # TODO: we perform force reboot which will fail in some cases. Need to handle with supports mixin.
+      restart_now
+    end
+
+    def powered_on_now?
+      # TODO(miha-plesko): we should rely on VMDB state instead contacting provider.
+      # Update implementation once we have event-driven targeted refresh implemented.
+      with_provider_object { |system| return system.PowerState.to_s.downcase == 'on' }
+    end
+
+    private
+
+    def mac_addresses(system)
+      macs = system.EthernetInterfaces.Members
+      (macs.map(&:MACAddress) + macs.map(&:PermanentMACAddress)).reject(&:blank?).uniq
+    end
```

Comment:
@agrare @gtanzillo @tadeboro sorry for a bit of delay, was on PTO. But here it is, the updated `mac_addresses` method implementation plus a unit test for it https://github.com/ManageIQ/manageiq-providers-redfish/pull/62/files#diff-13dd5981e599cb8a74a1a5132fc57d29R72. We usually don't test private methods but in this case if feels like we have to.

Reply:
:+1: thanks @miha-plesko